fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ArrayItemReference.java

```diff
@@ -0,0 +1,150 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions;
+
+import org.apache.doris.nereids.trees.expressions.typecoercion.ExpectsInputTypes;
+import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
+import org.apache.doris.nereids.types.ArrayType;
+import org.apache.doris.nereids.types.DataType;
+import org.apache.doris.nereids.types.coercion.AnyDataType;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * it is item from array, which used in lambda function
+ */
+public class ArrayItemReference extends NamedExpression implements ExpectsInputTypes {
+    protected final String name;
+    protected final ExprId exprId;
+
+    /** ArrayItemReference */
+    public ArrayItemReference(String name, Expression arrayExpression) {
+        this(StatementScopeIdGenerator.newExprId(), name, arrayExpression);
+    }
+
+    public ArrayItemReference(ExprId exprId, String name, Expression arrayExpression) {
+        super(ImmutableList.of(arrayExpression));
+        Preconditions.checkArgument(arrayExpression.getDataType() instanceof ArrayType,
+                "ArrayItemReference' child must return array");
+        this.name = name;
+        this.exprId = exprId;
+    }
+
+    public Expression getArrayExpression() {
+        return children.get(0);
+    }
+
+    @Override
+    public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
+        return visitor.visitArrayItemReference(this, context);
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public ExprId getExprId() {
+        return exprId;
+    }
+
+    @Override
+    public List<String> getQualifier() {
+        return ImmutableList.of(name);
+    }
+
+    @Override
+    public boolean nullable() {
+        return ((ArrayType) (this.children.get(0).getDataType())).containsNull();
+    }
+
+    @Override
+    public ArrayItemReference withChildren(List<Expression> expressions) {
+        return new ArrayItemReference(exprId, name, expressions.get(0));
+    }
+
+    @Override
+    public DataType getDataType() {
+        return ((ArrayType) (this.children.get(0).getDataType())).getItemType();
+    }
+
+    @Override
+    public String toSql() {
+        String str = getName() + "#" + getExprId();
+        str += " of " + child(0).toSql();
+        return str;
```

Comment:
to sql should return valid sql to ensure lamdba.toSql could return valid sql

Reply:
done